x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/querydsl/query/SingleValueQuery.java

```diff
@@ -210,14 +230,16 @@ Stats stats() {
     }
 
     private static class LuceneQuery extends org.apache.lucene.search.Query {
-        private final org.apache.lucene.search.Query next;
+        final org.apache.lucene.search.Query next;
         private final IndexFieldData<?> fieldData;
         private final Stats stats;
+        private final Warnings warnings;
 
-        LuceneQuery(org.apache.lucene.search.Query next, IndexFieldData<?> fieldData, Stats stats) {
```

Comment:
thought: maybe it's sufficient to hand over a `Source`, rather than expecting that the `Warnings` get contructed from outside?

Reply:
I wanted to have the `Warning` instantiation as high in the call flow as possible. Do you see a reasons to do otherwise?

Reply:
Argh, sorry, I missed the usage in `rewrite` - thought we were always handing the constructor a `new Warnings(source)`. My bad!